src/durable-cache/src/lib.rs

```diff
@@ -135,7 +135,9 @@ impl<C: DurableCacheCodec> DurableCache<C> {
             let events = self.subscribe.fetch_next().await;
             for event in events {
                 match event {
-                    ListenEvent::Updates(x) => {
+                    ListenEvent::Updates(mut x) => {
+                        consolidate_updates(&mut x);
+                        x.sort_by(|(_, ts1, d1), (_, ts2, d2)| ts1.cmp(ts2).then(d1.cmp(d2)));
```

Comment:
Why's this now needed? You noticed that something doesn't work right without it?

Reply:
Yes, the panic below on line 158 was triggered 

```Rust
assert_eq!(val, prev, "removed val does not match expected val");

```

We have to process retractions before inserts or else we may retract the wrong value.